### PR TITLE
ENH: Added schema kwarg to get_schema method

### LINKED ISSUE #28486: Add support for schema parameter when calling io.sql.get_schema
Currently, there is no way to specify the schema when calling pandas.io.sql.get_schema, the schema is a helpful parameter when working with certain databases such as MSSQL.

```python
import pandas.io.sql

# current state
table_name = 'my_new_table'
sql_create_statement = pandas.io.sql.get_schema(df, table_name, con)
# create statement looks like
"""
CREATE TABLE [my_new_table] (
    [column1] VARCHAR(max) NULL,
    ....
)
"""
```

this adds an extra step to have to parse the file in order to add the schema, otherwise when I execute the script later, the table is created in user default schema (dbo in most cases). This issue can be easily addressed by adding  `schema` parameter to `_create_sql_schema` since `SQLTable` already supports `schema`

```python

import pandas.io.sql

# future state
table_name = 'my_new_table'
schema = 'pypi'
sql_create_statement = pandas.io.sql.get_schema(df, table_name, schema=schema)

# create statement will look like this for MSSQL if con parameter is passed
"""
CREATE TABLE pypi.[my_new_table] (
    [Column1],
    ....
)
"""
```

### PR DESCRIPTION
**Pull request by gregorylivschitz**
- [x] closes #28486
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by alimcmaster1** [comment]
Please can you merge master

File: pandas/tests/io/test_sql.py [hunk 1]
```diff
@@ -855,6 +855,12 @@ def test_get_schema(self):
         create_sql = sql.get_schema(self.test_frame1, "test", con=self.conn)
         assert "CREATE" in create_sql
 
+    def test_get_schema_with_schema(self):
+        create_sql = sql.get_schema(
+            self.test_frame1, "test", con=self.conn, schema="pypi"
+        )
+        assert "pypi" in create_sql
```
Comment (alimcmaster1) [inline pandas/tests/io/test_sql.py:862]: Could we make the test a little more robust here. ie. check just the table name is prefixed with the schema

**Review by alimcmaster1 — CHANGES_REQUESTED** [review]

File: pandas/tests/io/test_sql.py (hunk 1 above)
Comment (gregorylivschitz) [reply pandas/tests/io/test_sql.py]: @alimcmaster1 
Thanks so much for reviewing my pr, I have changed the test case and rebased from master.

**Comment by alimcmaster1** [comment]
Seems like there are some test failures - mind taking a look?

**Comment by gregorylivschitz** [comment]
@alimcmaster1 

I'm not sure what the issue was, it was not my test that was breaking, I rebased from master again and it seems to be working now.

File: doc/source/whatsnew/v1.1.0.rst [hunk 2]
```diff
@@ -98,6 +98,8 @@ Other enhancements
   This can be used to set a custom compression level, e.g.,
   ``df.to_csv(path, compression={'method': 'gzip', 'compresslevel': 1}``
   (:issue:`33196`)
+- :meth:`get_schema` will now allow a schema kwarg that will add a schema into the create table statement
+-
```
Comment (alimcmaster1) [inline doc/source/whatsnew/v1.1.0.rst:102]: please add an issue number

File: pandas/tests/io/test_sql.py [hunk 3]
```diff
@@ -855,6 +855,13 @@ def test_get_schema(self):
         create_sql = sql.get_schema(self.test_frame1, "test", con=self.conn)
         assert "CREATE" in create_sql
 
+    def test_get_schema_with_schema(self):
+        create_sql = sql.get_schema(
+            self.test_frame1, "test", con=self.conn, schema="pypi"
+        )
+        print(create_sql)
```
Comment (alimcmaster1) [inline pandas/tests/io/test_sql.py:862]: can remove this print statement

**Review by alimcmaster1 — CHANGES_REQUESTED** [review]
Almost there - small comments

**Comment by gregorylivschitz** [comment]
@alimcmaster1 
I have completed the requests.

**Comment by alimcmaster1** [comment]
This looks good to me @gregorylivschitz mind merging master?

**Comment by jreback** [comment]
@gregorylivschitz can you merge master

**Comment by gregorylivschitz** [comment]
@jreback 
It is done.

**Review by alimcmaster1 — APPROVED** [review]
LGTM - @gregorylivschitz mind merging master?

**Comment by gregorylivschitz** [comment]
@alimcmaster1 @MarcoGorelli 

I have merged it in. Thanks for reminding me!

File: pandas/io/sql.py [hunk 4]
```diff
@@ -1455,9 +1455,15 @@ def drop_table(self, table_name, schema=None):
             self.get_table(table_name, schema).drop()
             self.meta.clear()
```
Comment (MarcoGorelli) [inline pandas/io/sql.py:1457]: While you're here, could you add type annotations to these arguments? Not all old code has them, but they're encouraged for new developments. Feel free to ping if it's not something you're familiar with

File: pandas/io/sql.py [hunk 5]
```diff
@@ -1588,9 +1594,13 @@ def _create_table_setup(self):
             create_tbl_stmts.append(
                 f"CONSTRAINT {self.name}_pk PRIMARY KEY ({cnames_br})"
             )
-
+        if self.schema:
+            schema_to_add = self.schema + "."
```
Comment (MarcoGorelli) [inline pandas/io/sql.py:1598]: Granted I'm no SQL expert, but perhaps `schema_name` would be a better variable name?

File: pandas/tests/io/test_sql.py [hunk 6]
```diff
@@ -866,6 +866,12 @@ def test_get_schema(self):
         create_sql = sql.get_schema(self.test_frame1, "test", con=self.conn)
         assert "CREATE" in create_sql
 
+    def test_get_schema_with_schema(self):
```
Comment (MarcoGorelli) [inline pandas/tests/io/test_sql.py:869]: can you add the github issue number here, e.g. `# GH8624` (but with the correct number)

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
@gregorylivschitz overall looks good to me, have just left some minor comments

File: pandas/io/sql.py (hunk 4 above)
Comment (MarcoGorelli) [inline pandas/io/sql.py:1457]: While you're here, could you add type annotations to these arguments? Not all old code has them, but they're encouraged for new developments. Feel free to ping if it's not something you're familiar with

Comment (MarcoGorelli) [inline pandas/io/sql.py:1457]: While you're here, could you add type annotations to these arguments? Not all old code has them, but they're encouraged for new developments. Feel free to ping if it's not something you're familiar with

File: pandas/io/sql.py [hunk 7]
```diff
@@ -1455,9 +1455,15 @@ def drop_table(self, table_name, schema=None):
             self.get_table(table_name, schema).drop()
             self.meta.clear()
 
-    def _create_sql_schema(self, frame, table_name, keys=None, dtype=None):
+    def _create_sql_schema(self, frame, table_name, keys=None, dtype=None, schema=None):
```
Comment (MarcoGorelli) [inline pandas/io/sql.py:1458]: While you're here, could you add type annotations to these arguments? Not all old code has them, but they're encouraged for new developments. Feel free to ping if it's not something you're familiar with

**Comment by WillAyd** [comment]
@gregorylivschitz I think this looks good but has a merge conflict and a CI failure - can you take a look? Should be able to get this in soon

**Review by arw2019 — COMMENTED** [review]
@gregorylivschitz if you merge master and resolve conflicts it sounds like this is good to go in

**Comment by arw2019** [comment]
Closing in favor of #38146

**Pull request closed without merging**